.github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml

```diff
@@ -0,0 +1,121 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Java Amazon-Web-Services IO Direct
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths:
+      - "sdks/java/io/amazon-web-services/**"
+      - "sdks/java/io/common/**"
+      - "sdks/java/core/src/main/**"
+      - "build.gradle"
+      - "buildSrc/**"
+      - "gradle/**"
+      - "gradle.properties"
+      - "gradlew"
+      - "gradle.bat"
+      - "settings.gradle.kts"
+      - ".github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml"
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths:
+      - "sdks/java/io/amazon-web-services/**"
+      - "sdks/java/io/common/**"
+      - "sdks/java/core/src/main/**"
+      - "build.gradle"
+      - "buildSrc/**"
+      - "gradle/**"
+      - "gradle.properties"
+      - "gradlew"
+      - "gradle.bat"
+      - "settings.gradle.kts"
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+jobs:
+  beam_PreCommit_Java_Amazon-Web-Services_IO_Direct:
+    name: beam_PreCommit_Java_Amazon-Web-Services_IO_Direct (Run Java_Amazon-Web-Services_IO_Direct PreCommit)
+    timeout-minutes: 60
+    if: |
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' || 
+      github.event.comment.body == 'Run Java_Amazon-Web-Services_IO_Direct PreCommit' || 
+      github.event_name == 'schedule'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Rerun on comment
+        if: github.event.comment.body == 'Run Java_Amazon-Web-Services_IO_Direct PreCommit'
+        uses: ./.github/actions/rerun-job-action
+        with:
+          pull_request_url: ${{ github.event.issue.pull_request.url }}
+          github_repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ github.job }}
+          github_current_run_id: ${{ github.run_id }}
```

Comment:
Please update these steps in accordance with recent changes:

```
- uses: actions/checkout@v3
- name: Setup repository
  uses: ./.github/actions/setup-action
  with:
    comment_phrase: 'Run Java_Amazon-Web-Services_IO_Direct PreCommit'
    github_token: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Done